src/core/Akka.Remote/Endpoint.cs

```diff
@@ -1100,23 +1030,27 @@ private IActorRef StartReadEndpoint(AkkaProtocolHandle handle)
         {
             var newReader =
                 Context.ActorOf(RARP.For(Context.System)
-                .ConfigureDispatcher(
-                    EndpointReader.ReaderProps(LocalAddress, RemoteAddress, Transport, Settings, _codec, _msgDispatcher,
-                        Inbound, (int)handle.HandshakeInfo.Uid, _receiveBuffers, _reliableDeliverySupervisor)
-                        .WithDeploy(Deploy.Local)),
+                    .ConfigureDispatcher(
+                        EndpointReader.ReaderProps(LocalAddress, RemoteAddress, Transport, Settings, _codec, _msgDispatcher,
+                            Inbound, (int)handle.HandshakeInfo.Uid, _receiveBuffers, _reliableDeliverySupervisor)
+                            .WithDeploy(Deploy.Local)),
                     string.Format("endpointReader-{0}-{1}", AddressUrlEncoder.Encode(RemoteAddress), _readerId.Next()));
             Context.Watch(newReader);
             handle.ReadHandlerSource.SetResult(new ActorHandleEventListener(newReader));
             return newReader;
         }
 
+        /// <summary>
+        /// TODO: this varies from the JVM but I can't tell if it's significant
```

Comment:
Ah, this could be a big issue.
You need to use 
`Serialization.SerializeWithTransport(system,address, () => return serializedObj..);`

That is our version of 

```scala
      Serialization.currentTransportInformation.withValue(Serialization.Information(h.localAddress, extendedSystem)) {
        (MessageSerializer.serialize(extendedSystem, msg.asInstanceOf[AnyRef]))
      }
```

Which sets the address information for the serializer so that actorrefs are serialized with the correct address from the local system.